flexmeasures/api/v3_0/tests/test_assets_api.py

```diff
@@ -366,3 +365,113 @@ def test_get_assets_new(
     print("Server responded with:\n%s" % get_assets_response.json)
     assert get_assets_response.status_code == 200
     assert len(get_assets_response.json) == 1
+    assert get_assets_response.json[0]["name"] == "Test ConsultantClient Asset"
+
+
+@pytest.mark.parametrize(
+    "requesting_user", ["test_consultant_user@seita.nl"], indirect=True
+)
+def test_consultant_can_not_patch(
+    client,
+    setup_api_test_data,
+    setup_accounts,
+    requesting_user,
+):
+    """
+    Try to edit an asset belonging to the ConsultantClient account with the Consultant account.
+    The Consultant account only has read access.
+    """
+    consultant_client_asset = GenericAsset.query.filter_by(
+        name="Test ConsultantClient Asset"
+    ).one_or_none()
+    print(consultant_client_asset)
+
+    asset_edit_response = client.patch(
+        url_for("AssetAPI:patch", id=consultant_client_asset.id),
+        json={
+            "latitude": consultant_client_asset.latitude,
+        },
+    )
+    print(f"Editing Response: {asset_edit_response.json}")
+    assert asset_edit_response.status_code == 403
+
+
+@pytest.mark.parametrize(
+    "requesting_user",
+    ["test_consultant_user_without_access@seita.nl"],
+    indirect=True,
+)
+def test_consultant_without_customer_manager_role(
+    client,
+    setup_api_test_data,
+    setup_accounts,
+    requesting_user,
+):
+    """
+    The Consultant Account user without customer manager role can not read.
+    """
+    account_name = "ConsultantClient"
+    query = {"account_id": setup_accounts[account_name].id}
+
+    get_assets_response = client.get(
+        url_for("AssetAPI:index"),
+        query_string=query,
+    )
+    print("Server responded with:\n%s" % get_assets_response.json)
+    assert get_assets_response.status_code == 403
+
+
+@pytest.mark.parametrize(
+    "parent_name, child_name, fails",
+    [
+        ("parent", "child_4", False),
+        (None, "child_1", False),
+        (None, "child_1", True),
+        ("parent", "child_1", True),
+    ],
+)
+@pytest.mark.parametrize("requesting_user", ["test_admin_user@seita.nl"], indirect=True)
```

Comment:
Something went wrong when merging, because this test comes from [another PR](https://github.com/FlexMeasures/flexmeasures/pull/874/files#diff-045dbe938b968f5c33e77033a57045cdcf6e24f77b7b466405aa391ecdff624eR344).

So this should go.

I'm not sure if there are more like that, can you check?

Reply:
Thanks for pointing this out. Apparently it is correct but good to check the merge because it was indeed a confusing one. This line is also in line 353 of the other PR.